docs/deployment/trimming.md

```diff
@@ -0,0 +1,172 @@
+
+To prevent changes in behavior when trimming apps, .NET provides static analysis of trim compatibility through trim warnings. The trimmer produces trim warnings when it finds code that might not be compatible with trimming. If there are any trim warnings, the app should be thoroughly tested after trimming to ensure that there are no behavior changes. For more information, see [Introduction to trim warnings](/dotnet/core/deploying/trimming/fixing-warnings).
+
+## Trimming behavior
+
+Trimming behavior can be controlled by setting the `$(TrimMode)` build property to either `partial` or `full`:
+
+```xml
+<TrimMode>full</TrimMode>
+```
+
+> [!IMPORTANT]
+> The `$(TrimMode)` build property shouldn't be conditioned by build configuration. This is because features switches are enabled or disabled based on the value of the `$(TrimMode)` build property, and the same features should be enabled or disabled in all build configurations so that your code behaves identically.
+
+The `full` trim mode removes any code that's not used by your app. The `partial` trim mode trims SDK assemblies, and any other assemblies that have opted into trimming with the `$(TrimmableAsssembly)` MSBuild item:
+
+```xml
+<ItemGroup>
+  <TrimmableAssembly Include="MyAssembly" />
+</ItemGroup>
+```
+
+> [!NOTE]
+> This is equivalent to setting `[AssemblyMetadata("IsTrimmable", "True")]` when building the assembly.
+
+For more trimming options, see [Trimming options](/dotnet/core/deploying/trimming/trimming-options).
+
+## Trimming defaults
+
+By default, Android and Mac Catalyst builds use partial trimming when the build configuration is set to a release build. iOS uses partial trimming for any device builds, regardless of the build configuration, and doesn't use trimming for simulator builds.
+
+## Trimming incompatibilities
+
+[!INCLUDE [Trimming incompatibilities](includes/trimming-incompatibilities.md)]
+
+Alternatively, you can use feature switches so that the trimmer preserves the code for these features. For more information, see [Trimming feature switches](#trimming-feature-switches).
+
+For .NET trimming incompatibilities, see [Known trimming incompatibilities](/dotnet/core/deploying/trimming/incompatibilities).
+
+### Define a TypeConverter to replace an implicit conversion operator
+
+Implicit conversion operators in a .NET MAUI app are incompatible with full trimming and will be removed by the trimmer. In a .NET MAUI app these typically affect bindings between properties with different types, and setting a property value of a bindable object with a value of a different type. For more information about implicit conversion operators, see [User-defined explicit and implicit conversion operators](/dotnet/csharp/language-reference/operators/user-defined-conversion-operators).
```

Comment:
> Implicit conversion operators in a .NET MAUI app

I'm not sure how I feel about this statement. The problem is that when we're evaluating XAML at runtime (especially with data bindings), we might not be able to convert between two types using the implicit operators because they can be trimmed. Implicit operators are a C# feature and not a .NET feature. I just don't know if there's a good way to describe the problem so that it isn't more confusing than helpful.

Maybe something like this could work:

"It is not possible to rely on implicit conversion operators when assigning a value of incompatible type to a property in XAML or when two properties of different types are connected with a data binding when full trimming is enabled. The implicit operator methods might be removed by the trimmer if they aren't used in your C# code."